drivers/gps/nrf9160_gps/nrf9160_gps.c

```diff
@@ -551,6 +554,34 @@ static int start(struct device *dev, struct gps_config *cfg)
 		return -EIO;
 	}
 
+	if (gps_cfg.priority) {
+		retval = nrf_setsockopt(drv_data->socket,
+					NRF_SOL_GNSS,
+					NRF_SO_GNSS_ENABLE_PRIORITY, NULL, 0);
+		if (retval != 0) {
+			LOG_ERR("Failed to enable GPS priority");
+			return -EIO;
+		}
+	} else {
+		retval = nrf_setsockopt(drv_data->socket,
+					NRF_SOL_GNSS,
+					NRF_SO_GNSS_DISABLE_PRIORITY, NULL, 0);
+		if (retval != 0) {
+			LOG_ERR("Failed to disable GPS priority");
+			return -EIO;
+		}
+	}
+
+	retval = nrf_setsockopt(drv_data->socket,
+				NRF_SOL_GNSS,
+				NRF_SO_GNSS_START,
```

Comment:
`NRF_SO_GNSS_START` option is already being set in line 547, shouldn't that `nrf_setsockopt()` call be removed?

Reply:
It's actually an intentional workaround for an issue in BSDlib. `NRF_SO_GNSS_START` must be set for `NRF_SO_GNSS_ENABLE_PRIORITY` / `NRF_SO_GNSS_DISABLE_PRIORITY` to not return an error. At the same time, setting one of the two latter implicitly stops the GPS, so it has to be started again.

Reply:
Could you add a comment clarifying that? It's a bit confusing to see that GPS is started twice, seems unintentional.

Reply:
Will do

Reply:
Added comments in the last force-push.